File: Atmosphere.BE/Atmosphere.Core/Enums/ReadingType.cs
using System.Runtime.Serialization;

namespace Atmosphere.Core.Enums;

public enum ReadingType
{
    [EnumMember(Value = "unknown")] Unknown = -1,

    [EnumMember(Value = "temperature")] Temperature = 0,

    [EnumMember(Value = "humidity")] Humidity = 1,

    [EnumMember(Value = "pressure")] Pressure = 2
}

File: Atmosphere.RPI/lib/Models/ReadingType.cpp


#include "ReadingType.h"

using namespace Tiny;

std::string Tiny::ReadingTypeToString(ReadingType value) {
  switch (value) {
    case ReadingType::Unknown:
      return "unknown";
    case ReadingType::Temperature:
      return "temperature";
    case ReadingType::Humidity:
      return "humidity";
    case ReadingType::Pressure:
      return "pressure";
    default:
      return "Unknown";
  }
}

ReadingType Tiny::ReadingTypeFromString(std::string value) {
  if (value == "unknown") {
    return ReadingType::Unknown;
  }
  if (value == "temperature") {
    return ReadingType::Temperature;
  }
  if (value == "humidity") {
    return ReadingType::Humidity;
  }
  if (value == "pressure") {
    return ReadingType::Pressure;
  }
  return ReadingType::Unknown;
}
